subsys/bluetooth/audio/shell/cap_initiator.c

```diff
@@ -481,14 +484,676 @@ static int cmd_cap_initiator_unicast_cancel(const struct shell *sh, size_t argc,
+
+				return -ENOEXEC;
+			}
+			snk_ep_cnt++;
+		}
+#endif /* UNICAST_SINK_SUPPORTED */
+
+#if UNICAST_SRC_SUPPORTED
+		for (size_t j = 0U; j < param->src_cnt[i]; j++) {
+			src_eps[src_ep_cnt] = srcs[bt_conn_index(connected_conns[i])][j];
+			if (src_eps[src_ep_cnt] == NULL) {
+				shell_error(ctx_shell, "No source[%zu][%zu] endpoint available", i,
+					    j);
+
+				return -ENOEXEC;
+			}
+			src_ep_cnt++;
+		}
+#endif /* UNICAST_SRC_SUPPORTED  */
+	}
+
+	if (snk_ep_cnt != snk_cnt) {
+		shell_error(ctx_shell, "Sink endpoint and stream count mismatch: %zu != %zu",
+			    snk_ep_cnt, snk_cnt);
+
+		return -ENOEXEC;
+	}
+
+	if (src_ep_cnt != src_cnt) {
+		shell_error(ctx_shell, "Source  endpoint and stream count mismatch: %zu != %zu",
+			    src_ep_cnt, src_cnt);
+
+		return -ENOEXEC;
+	}
+
+	/* Setup arrays of parameters based on the preset for easier access. This also copies the
+	 * preset so that we can modify them (e.g. update the metadata)
+	 */
+	for (size_t i = 0U; i < snk_cnt; i++) {
+		snk_cap_streams[i] = &snk_uni_streams[i]->stream;
+		snk_qos[i] = &snk_uni_streams[i]->qos;
+		snk_codec_cfgs[i] = &snk_uni_streams[i]->codec_cfg;
+	}
+
+	for (size_t i = 0U; i < src_cnt; i++) {
+		src_cap_streams[i] = &src_uni_streams[i]->stream;
+		src_qos[i] = &src_uni_streams[i]->qos;
+		src_codec_cfgs[i] = &src_uni_streams[i]->codec_cfg;
+	}
+
+	/* CAP Start */
+	for (size_t i = 0U; i < param->conn_cnt; i++) {
+		for (size_t j = 0U; j < param->snk_cnt[i]; j++) {
+			struct bt_cap_unicast_audio_start_stream_param *stream_param =
+				&stream_params[stream_cnt];
+
+			stream_param->member.member = connected_conns[i];
+			stream_param->codec_cfg = snk_codec_cfgs[snk_stream_cnt];
+			stream_param->ep = snk_eps[snk_stream_cnt];
+			stream_param->qos = snk_qos[snk_stream_cnt];
+			stream_param->stream = snk_cap_streams[snk_stream_cnt];
+
+			snk_stream_cnt++;
+			stream_cnt++;
+
+			if (param->conn_cnt > 1) {
+				const int err =
+					codec_set_chan_alloc(stream_param->codec_cfg, BIT(i));
+
+				if (err != 0) {
+					shell_error(ctx_shell,
+						    "Failed to set channel allocation for "
+						    "snk[%zu][%zu]: %d",
+						    i, j, err);
+
+					return err;
+				}
+			}
+		}
+
+		for (size_t j = 0U; j < param->src_cnt[i]; j++) {
+			struct bt_cap_unicast_audio_start_stream_param *stream_param =
+				&stream_params[stream_cnt];
+
+			stream_param->member.member = connected_conns[i];
+			stream_param->codec_cfg = src_codec_cfgs[src_stream_cnt];
+			stream_param->ep = src_eps[src_stream_cnt];
+			stream_param->qos = src_qos[src_stream_cnt];
+			stream_param->stream = src_cap_streams[src_stream_cnt];
+
+			src_stream_cnt++;
+			stream_cnt++;
+
+			if (param->conn_cnt > 1) {
+				const int err =
+					codec_set_chan_alloc(stream_param->codec_cfg, BIT(i));
+
+				if (err != 0) {
+					shell_error(ctx_shell,
+						    "Failed to set channel allocation for "
+						    "src[%zu][%zu]: %d",
+						    i, j, err);
+
+					return err;
+				}
+			}
```

Comment:
Do we need UNICAST SRC/SNK SUPPORTED guards here as well?

Reply:
Shouldn't be necessary. 

The checks in 
```c
	if (snk_ep_cnt != snk_cnt) {
		shell_error(ctx_shell, "Sink endpoint and stream count mismatch: %zu != %zu",
			    snk_ep_cnt, snk_cnt);

		return -ENOEXEC;
	}

	if (src_ep_cnt != src_cnt) {
		shell_error(ctx_shell, "Source  endpoint and stream count mismatch: %zu != %zu",
			    src_ep_cnt, src_cnt);

		return -ENOEXEC;
	}
```

Should ensure correctness before we reach this point. The reason why the above was guarded, is that `srcs` and `snks` arrays are only conditionally compiled (i.e. guarded). 

So if attempting to do something with sinks when not supported would result in a runtime check